src/components/ToDoItemComp.vue

```diff
@@ -1,32 +1,223 @@
+              </li>
+            </ul>
+          </div>
+        </div>
+      </div>
+      <div
+        class="flex flex-row space-x-1"
+        :class="showDropDown ? 'opacity-30' : ''"
+      >
+        <img src="../assets/icons/Vectorcalendar.svg" />
+        <p class="text-xs">{{ formatShortDate(date) }}</p>
+      </div>
+      <div class="self-start mt-4 mb-4">
+        <textarea
+          v-model="currentText"
+          placeholder="Lorem ipsum"
+          class="border-none h-[70.27px] w-[224.45px] lg:h-[103px] lg:w-[455.1px] text-start text-sm lg:text-lg text-grey"
+          :class="showDropDown ? 'opacity-30' : ''"
+          :disabled="showDropDown"
+        ></textarea>
+      </div>
+      <div class="text-sm flex flex-row font-semibold space-x-2">
+        <button
+          class="text-white w-[61px] h-[27px] rounded-lg bg-green"
+          :class="showDropDown ? 'opacity-30' : ''"
+          :disabled="showDropDown"
+          @click="saveToDo"
+        >
+          Save
+        </button>
+        <button
+          class="rounded-lg h-[27px] w-[70px] bg-buttonGray"
+          :class="showDropDown ? 'opacity-30' : ''"
+          :disabled="showDropDown"
+          @click="deleteToDo"
+        >
+          Delete
+        </button>
+      </div>
     </div>
   </div>
 </template>
 
 <script setup lang="ts">
-  import { ToDoItem } from '../models/todoitem-model';
+  import { computed, ref } from 'vue';
+  import { Priority, ToDoItem, Status } from '../models/todoitem-model';
+  import { formatShortDate } from '../utils/date-formatting';
+  import { getItemPriority } from '../utils/item-priority';
 
   const props = defineProps<{
-    item: ToDoItem;
-    index: number;
+    item?: ToDoItem;
+    index?: number;
   }>();
 
   const emit = defineEmits<{
     (e: 'removeItem', ind: number): void;
+    (e: 'addToDo', newToDo: ToDoItem): void;
+    (e: 'hideForm'): void;
+    (e: 'updateToDo', ind: number, newToDo: ToDoItem): void;
   }>();
 
-  function removeTodo(ind: number) {
-    emit('removeItem', ind);
+  const showDropDown = ref(false);
+
+  const currentTitle = ref(props.item ? props.item.title : '');
+
+  const currentText = ref(props.item ? props.item.text : '');
+
+  const indexOfThisItem = ref(
+    props.index || props.index === 0 ? props.index : -1
+  );
+
+  const currentPriority = ref(props.item ? props.item.priority : Priority.low);
+
+  const priorityClass = computed(() => getItemPriority(currentPriority.value));
+
+  const date = new Date();
+
+  const priorityList = [Priority.low, Priority.medium, Priority.high];
+
+  const showPopUp = ref(false);
+
+  function changePriority(priority: Priority) {
+    currentPriority.value = priority;
+    showDropDown.value = false;
+  }
+
+  function saveToDo() {
+    const newToDo = {
+      title: currentTitle.value,
+      text: currentText.value,
+      createdAt: new Date(),
+      priority: currentPriority.value,
+      status: Status.notresolved,
```

Comment:
`notresolved` is this a boolean value? If so, please change rename it use `is` prefix and camelCase as well.